openlibrary/core/vendors.py

```diff
@@ -43,92 +44,83 @@ def search_amazon(title='', author=''):
-        price, qlt = (used, 'used') if used else (new, 'new')
-
-    if price:
-        price = '{:00,.2f}'.format(int(price)/100.)
-        if qlt:
-            price_fmt = "$%s (%s)" % (price, qlt)
-
-    data = {
+    {
+      'price': '$54.06',
+      'price_amt': 5406,
+      'binding': 'Hardcover',
+      'authors': {'Greenfield, Ben': 'Author'},
+      'publication_date': '2020-01-21T00:00:01Z',
+      'dimensions': {
+        'width': [1.7, 'Inches'],
+        'length': [8.5, 'Inches'],
+        'weight': [5.4, 'Pounds'],
+        'height': [10.875, 'Inches']
+       },
+       'publisher': 'Victory Belt Publishing',
+       'source_records': ['amazon:1628603976'],
+       'title': 'Boundless: Upgrade Your Brain, Optimize Your Body & Defy Aging',
+       'url': 'https://www.amazon.com/dp/1628603976/?tag=internetarchi-20',
+       'number_of_pages': 640,
+       'cover': 'https://m.media-amazon.com/images/I/51IT9MV3KqL._AC_.jpg',
+       'languages': {'Original Language': 'English', 'Published': 'English'},
+       'edition_num': '1'
+     }
+    """
+    if not product:
+        return {}  # no match?
+
+    item_info = product.raw_info.item_info
+    edition_info = item_info.content_info
+    attribution = item_info.by_line_info
+    price = product.prices and product.prices.price
+    dims = item_info.product_info and item_info.product_info.item_dimensions
+    book = {
         'url': "https://www.amazon.com/dp/%s/?tag=%s" % (
             product.asin, h.affiliate_id('amazon')),
-        'price': price_fmt,
-        'price_amt': price,
-        'qlt': qlt,
-        'title': product.title,
-        'authors': [{'name': name} for name in product.authors],
         'source_records': ['amazon:%s' % product.asin],
-        'number_of_pages': product.pages,
-        'languages': list(product.languages),
-        'cover': product.large_image_url,
-        'product_group': product.product_group,
+        'isbn_10': [product.asin],
+        'isbn_13': [isbn_10_to_isbn_13(product.asin)],
+        'price': price and price.display,
+        'price_amt': price.value and int(100 * price.value),
+        'title': product.title,
+        'cover': product.images and product.images.large,
+        'authors': [{'name': contrib.name, 'role': contrib.role}
+                    for contrib in attribution.contributors],
+        'publishers': attribution.brand and [attribution.brand.display_value],
+        'number_of_pages': (edition_info.pages_count and
+                            edition_info.pages_count.display_value),
+        'edition_num': (edition_info.edition and
+                        edition_info.edition.display_value),
+        'publish_date': (edition_info.publication_date and
+                             edition_info.publication_date.display_value),
+        'languages': (
+            edition_info.languages and
+            list(set(lang.display_value
+                     for lang in edition_info.languages.display_values))
```

Comment:
Does the API give language codes of any sort? We use 3 character bibliographic language codes, _not_ names. If the codes are available from the API, that would be ideal.